libraries/esp8266/examples/GPIO/FunctionalInterrupt/FunctionalInterrupt.ino

```diff
@@ -0,0 +1,85 @@
+#include <Arduino.h>
+#include <Schedule.h>
+#include "FunctionalInterrupts.h"
+
+#if defined(ESP32)
+#define BUTTON1 16
+#define BUTTON2 17
+#elif defined(ARDUINO_ESP8266_WEMOS_D1MINI)
+#define BUTTON1 D4
+#define BUTTON2 D3
+#else
+#define BUTTON1 2
+#define BUTTON2 0
+#endif
+
+class Button {
+  public:
+    Button(uint8_t reqPin) : PIN(reqPin) {
+      pinMode(PIN, INPUT_PULLUP);
+      // Arduino C API:
+      attachInterruptArg(PIN, [](void* self) {
+        static_cast<Button*>(self)->isr();
+      }, this, FALLING); // works on ESP32; fails on ESP8266: "ISR not in IRAM"
```

Comment:
Any use of std::bind, std::function or lambdas in the code-path of ISRs in fact is a case of ISR not in IRAM, just like this, only wrapped in some other functions.